app/src/main/java/org/greenstand/android/TreeTracker/messages/ChatScreen.kt

```diff
@@ -0,0 +1,286 @@
+                .nestedScroll(scrollBehavior.nestedScrollConnection)
+            ) {
+                Messages(
+                    messages = state.messages,
+                    modifier = Modifier.weight(1f),
+                    scrollState = scrollState
+                )
+            Box(modifier = Modifier.padding(top = 4.dp, start = 4.dp, end = 4.dp ,bottom = 80.dp).fillMaxWidth().wrapContentHeight()){
+                TextField(
+                    modifier =Modifier.fillMaxWidth().padding(5.dp),
+                    value = state.draftText,
+                    onValueChange = { text -> viewModel.updateDraftText(text) },
+                    keyboardOptions = KeyboardOptions(
+                        keyboardType = KeyboardType.Text,
+                        imeAction = ImeAction.Go,
+                        autoCorrect = false,
+                    ),
+                    placeholder = {
+                        Text(
+                            text = stringResource(id = R.string.click_to_write_message),
+                            color = Color.White
+                        )
+                    },
+                    keyboardActions = KeyboardActions(
+                    onGo = {
+                        viewModel.sendMessage()
+                    }
+                ),
+                colors = TextFieldDefaults.textFieldColors(
+                    textColor = AppColors.LightGray,
+                    backgroundColor = AppColors.DeepGray,
+                    focusedIndicatorColor = Color.Transparent,
+                    unfocusedIndicatorColor = Color.Transparent,
+                    disabledIndicatorColor = Color.Transparent
+                ),
+                    maxLines = 2
+                )
+            }
+    }
+}
+}
+
+const val ConversationTestTag = "ConversationTestTag"
+
+@Composable
+fun Messages(
+    messages: Flow<List<DirectMessage>>?,
+    scrollState: LazyListState,
+    modifier: Modifier = Modifier
+) {
+    val scope = rememberCoroutineScope()
+    Box(modifier = modifier) {
+        val authorAdmin = "admin"
+        LazyColumn(
+            reverseLayout = true,
+            state = scrollState,
+            contentPadding = PaddingValues(top = 10.dp),
+            // Add content padding so that the content can be scrolled (y-axis)
+            // below the status bar + app bar
+            // TODO: Get height from somewhere
+            modifier = Modifier
+                .testTag(ConversationTestTag)
+                .fillMaxSize()
+        ) {
+            messages?.map { message ->
+                for (index in message.indices) {
+                    val prevAuthor = message.getOrNull(index - 1)?.from
+                    val nextAuthor = message.getOrNull(index + 1)?.from
+                    val content = message[index]
+                    val isFirstMessageByAuthor = prevAuthor != content.from
+                    val isLastMessageByAuthor = nextAuthor != content.from
```

Comment:
You can move this logic into the viewModel. Then the viewmodel can expose a custom data class just for this screen such as "ChatMessage" that has everything ready for the view to just render